File: Lab 2/lab2.py


step_count = 0

num = int(input("give bigger number than 9: "))

if num > 9:
    print(num, end="")

    while num > 9:
        current_sum = 0
        temp = num

        while temp > 0:
            current_sum += temp % 10
            temp //= 10

        num = current_sum
        step_count += 1
        print(f" -> {num}", end="")

    print(f"\n Final value: {num}")
    print(f"Total step: {step_count}")

else:
    print(f"{num} is not bigger than 9")

fizz_count = 0
buzz_count = 0
fizzbuzz_count = 0

n = int(input("Enter an integer between 10 and 100: "))

while n < 10 or n > 100:
    n = int(input("Invalid input. Enter a value between 10 and 100: "))

for i in range(1, n + 1):
    if i % 7 == 0:
        print(f"({i} is skipped)")
        continue

    if i % 3 == 0 and i % 5 == 0:
        print("FizzBuzz")
        fizzbuzz_count += 1
    elif i % 3 == 0:
        print("Fizz")
        fizz_count += 1
    elif i % 5 == 0:
        print("Buzz")
        buzz_count += 1
    else:
        print(i)

print("\n--- Summary ---")
print(f"Fizz: {fizz_count}")
print(f"Buzz: {buzz_count}")
print(f"FizzBuzz: {fizzbuzz_count}")

import sys

num1 = int(input("Enter a number between 3 and 9: "))

if num1 < 3 or num1 > 9:
    print("give valid input")
    sys.exit(0)

for i in range(1, 2 * num1):
    if i <= num1:
        k = i
    else:
        k = 2 * num1 - i

    for j in range(1, k + 1):
        print(f"{j} ", end="")
    print()

File: Lab 2/lab2.cpp
#include <iostream>

using namespace std;

int main() {
    //Task 1

    int num;
    int stepCount = 0;

    cout << "give bigger number than 9 ";
    cin >> num;

    if (num > 9) {
        cout << num;

        while (num > 9) {
            int currentSum = 0;
            int temp = num;

            while (temp > 0) {
                currentSum += temp % 10;
                temp /= 10;
            }

            num = currentSum;
            stepCount++;
            cout << " -> " << num;
        }

        cout << "\n Final value: " << num << endl;
        cout << "Total step: " << stepCount << endl;

    } else {
        cout << num << " is not bigger than 9" << endl;
    }


    //TASK 2
    int n;
    int fizzCount = 0;
    int buzzCount = 0;
    int fizzbuzzCount = 0;

    cout << "Enter an integer between 10 and 100: ";
    cin >> n;

    while (n < 10 || n > 100) {
        cout << "Invalid input. Enter a value between 10 and 100: ";
        cin >> n;
    }



    for (int i = 1; i <= n; i++) {
        if (i % 7 == 0) {
            cout << "("<< i << " is skipped" << ")" << endl;
            continue;
        }

        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizzbuzzCount++;
        }
        else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fizzCount++;
        }
        else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buzzCount++;
        }
        else {
            cout << i << endl;
        }
    }

    cout << "\n--- Summary ---" << endl;
    cout << "Fizz: " << fizzCount << endl;
    cout << "Buzz: " << buzzCount << endl;
    cout << "FizzBuzz: " << fizzbuzzCount << endl;


    //Task 3
    int num1;

    cout << "Enter a number between 3 and 9: ";
    cin >> num1;

    if (num1 < 3 || num1 > 9) {
        cout << "give valid input\n";
        return 0;
    }

    for (int i = 1; i <= 2 * num1 - 1; i++) {

        int k;
        if (i <= num1) {
            k = i;
        } else {
            k = 2 * num1 - i;
        }

        for (int j = 1; j <= k; j++) {
            cout << j << " ";
        }
        cout << "\n";
    }

    return 0;
}